diff --git a/hw/block/nvme.c b/hw/block/nvme.c
index 5fe082e..da95fa5 100644
--- a/hw/block/nvme.c
+++ b/hw/block/nvme.c
@@ -5580,6 +5580,10 @@ static void nvme_write_bar(NvmeCtrl *n, hwaddr offset, uint64_t data,
                        "invalid write to PMRCAP register, ignored");
         return;
     case 0xE04: /* PMRCTL */
+        if (!NVME_CAP_PMRS(n->bar.cap)) {
+            return;
+        }
+
         n->bar.pmrctl = data;
         if (NVME_PMRCTL_EN(data)) {
             memory_region_set_enabled(&n->pmr.dev->mr, true);
