diff --git a/hw/net/lan9118.c b/hw/net/lan9118.c
index 456ae38..47ab8ae 100644
--- a/hw/net/lan9118.c
+++ b/hw/net/lan9118.c
@@ -33,7 +33,7 @@
 #define DPRINTF(fmt, ...) \
 do { printf("lan9118: " fmt , ## __VA_ARGS__); } while (0)
 #define BADF(fmt, ...) \
-do { hw_error("lan9118: error: " fmt , ## __VA_ARGS__);} while (0)
+do {/* hw_error("lan9118: error: " fmt , ## __VA_ARGS__); */} while (0)
 #else
 #define DPRINTF(fmt, ...) do {} while(0)
 #define BADF(fmt, ...) \
@@ -1201,7 +1201,7 @@ static void lan9118_16bit_mode_write(void *opaque, hwaddr offset,
         return;
     }
 
-    hw_error("lan9118_write: Bad size 0x%x\n", size);
+    // hw_error("lan9118_write: Bad size 0x%x\n", size);
 }
 
 static uint64_t lan9118_readl(void *opaque, hwaddr offset,
@@ -1316,7 +1316,7 @@ static uint64_t lan9118_16bit_mode_read(void *opaque, hwaddr offset,
         return lan9118_readl(opaque, offset, size);
     }
 
-    hw_error("lan9118_read: Bad size 0x%x\n", size);
+    // hw_error("lan9118_read: Bad size 0x%x\n", size);
     return 0;
 }
 
